functions/activate-batch-background.js

```diff
@@ -65,14 +81,18 @@ async function handle(data, db, contract) {
     console.log('### Address shuffle round 3 ###')
     shuffledAddresses = shuffle(shuffledAddresses)
 
-    // Slice list for extra addresses
-    shuffledAddresses = shuffledAddresses.slice(0, prizes.length)
-
-    // Assign prizes based on addresses list shuffle
     let prizeAssignment = {}
     for(let i = 0;i < shuffledAddresses.length;i++) {
-      const val = prizeAssignment[shuffledAddresses[i]]
-      prizeAssignment[shuffledAddresses[i]] = [...(val ? val : []), prizes[i].name]
+      if (i < prizes.length) {
+        // Assign prizes based on addresses list shuffle. These are winners
+        const val = prizeAssignment[shuffledAddresses[i]]
+        prizeAssignment[shuffledAddresses[i]] = [...(val ? val : []), prizes[i].name]
+        // Increase badluck count for winners, in proportion to their number of nanopasses
+        addressBadluckCount[shuffledAddresses[i]].badLuckCount = Math.max(0, addressBadluckCount[shuffledAddresses[i]].badLuckCount - Math.ceil((addressBadluckCount[shuffledAddresses[i]].badLuckCount / holderBalance[shuffledAddresses[i]])));
+      } else {
+        // Increase badluck count for non-winners
+        addressBadluckCount[shuffledAddresses[i]].badLuckCount = addressBadluckCount[shuffledAddresses[i]].badLuckCount + 1;
+      }
```

Comment:
Ok I guess it can be done here, I was originally thinking of doing it in "open-box" on the website side, incase some people don't open the box, then we wouldn't modify things.

But here is good since open source (can verify algo) + all in one place.